src/main/java/com/redhat/devtools/lsp4ij/console/actions/ApplyLanguageServerSettingsAction.java

```diff
@@ -41,11 +51,55 @@ public void actionPerformed(@NotNull AnActionEvent e) {
 
     @Override
     public void update(@NotNull AnActionEvent e) {
-        e.getPresentation().setEnabled(languageServerView.isModified());
+        boolean modified = languageServerView.isModified();
+        boolean isShown = languageServerView.isSaveTipShown();
+
+        // Only show the tooltip once per configuration change
+        if (modified && !isShown) {
+            showBalloon();
+        }
+        // If configuration is returned to match the unmodified state, reset the tooltip to be shown again
+        if (!modified) {
+            languageServerView.isSaveTipShown(false);
+        }
+
+        e.getPresentation().setEnabled(modified);
     }
 
     @Override
     public @NotNull ActionUpdateThread getActionUpdateThread() {
         return ActionUpdateThread.BGT;
     }
+
+    private void showBalloon() {
+        languageServerView.isSaveTipShown(true);
+        JBLabel jbPanel = new JBLabel(LanguageServerBundle.message("action.lsp.detail.apply.balloon"));
+
+        BalloonBuilder builder = JBPopupFactory.getInstance()
+                .createBalloonBuilder(jbPanel)
+                .setFadeoutTime(800)
+                .setHideOnAction(false);
+
+        Balloon balloon = builder.createBalloon();
+
+        ActionToolbarImpl actionToolbarComponent = null;

```

Comment:
I will create a method which contains this code https://github.com/redhat-developer/lsp4ij/blob/d60f8f1f50b64e0c1af03cd9aa426d0d25c53bdd/src/main/java/com/redhat/devtools/lsp4ij/console/actions/ApplyLanguageServerSettingsAction.java#L85-L96 and which could return null (add `@Nullable`) in your method return and I will add some Javadoc which explains that the method search the Apply component

Reply:
Will you do that on main and I'll rebase before finishing this?

Reply:
Sorry, I wanted to say "please add Javadoc"